src/mds/QuiesceDbManager.cc

```diff
@@ -74,8 +74,24 @@ void* QuiesceDbManager::quiesce_db_thread_main()
 
     auto db_age = db.get_age();
 
-    if (!db_thread_has_work() && next_event_at_age > db_age) {
-      submit_condition.wait_for(ls, next_event_at_age - db_age);
+    // QuiesceTimeInterval::max() may overflow inside the call to
+    // std::condition_variable::wait_for and result in a busy-loop,
+    // making the call to timeout immediately
+    // The solution is to cap the wait duration to a value which can
+    // certainly fit in whichever clock std library is using internally.
+    while (!db_thread_has_work() && next_event_at_age > db_age) {
+      using namespace std::chrono_literals;
+      using std::chrono::duration_cast;
+      const auto max_wait = duration_cast<QuiesceTimeInterval>(10s);
+
+      auto wait_result = submit_condition.wait_for(
+        ls,
+        std::min(max_wait, next_event_at_age - db_age));
+
+      if (std::cv_status::timeout == wait_result) {
+        db_age = db.get_age();
```

Comment:
It seems to me that you should unconditionally update `db_age` no matter the result.